### PR TITLE
chore(sdk): release KFP SDK 1.8.16

### PR DESCRIPTION
**Pull request by connor-mccarthy**
**Description of your changes:**
Bumps version and updates release notes.

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn 
more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: sdk/RELEASE.md [hunk 1]
```diff
@@ -11,6 +11,21 @@
...
+## Breaking Changes
+
+### For Pipeline Authors
+
+### For Component Authors
+
+## Deprecations
+
+## Bug Fixes and Other Changes
+* Relax required google-auth version [#8470](https://github.com/kubeflow/pipelines/pull/8470)
```
Comment (chensun) [inline sdk/RELEASE.md:27]: nit: can be more specific, e.g.: `Require google-api-core>=1.31.5, >=2.3.2 [#7377](https://github.com/kubeflow/pipelines/pull/7377)`

**Review by chensun — COMMENTED** [review]
/lgtm

File: sdk/RELEASE.md (hunk 1 above)
Comment (connor-mccarthy) [reply sdk/RELEASE.md]: Good thinking. Updated.

**Review by chensun — APPROVED** [review]
/lgtm
/approve

Thanks!

**Pull request merged**